moodler/upload.py

```diff
@@ -0,0 +1,252 @@
+import re
+import logging
+import os
+import requests
+
+from typing import Tuple
+from moodler.assignment import Assignment
+
+from moodler.moodle_exception import MoodlerException
+from moodler.config import TOKEN, URL
+
+logger = logging.getLogger(__name__)
+
+GRADING_WORKSHEET_UPLOAD_CONTEXTID_PATTERN = (
+    r'<input name="contextid" type="hidden" ' r'value="(\d+)"'
+)
+
+GRADING_WORKSHEET_UPLOAD_SESSKEY_PATTERN = (
+    r'<input name="sesskey" type="hidden" ' r'value="([\w\d]+)"'
+)
+
+GRADING_WORKSHEET_UPLOAD_GRADESFILE_PATTERN = (
+    r'<input type="hidden" name="gradesfile" id="id_gradesfile" ' r'value="(\d+)"'
+)
+
+GRADING_WORKSHEET_UPLOAD_IMPORTID_PATTERN = (
+    r'<input name="importid" type="hidden" ' r'value="(\d+)"'
+)
+
+GRADING_WORKSHEET_UPLOAD_DRAFTID_PATTERN = (
+    r'<input name="draftid" type="hidden" ' r'value="(\d+)"'
+)
+
+SUCCESS_GRADING_WORKSHEET_UPLOAD_PATTERN = r"Updated (\d+) grades and feedback"
+
+
+class UploadException(MoodlerException):
+    pass
+
+
+class InvalidGradingPage(UploadException):
+    pass
+
+
+class InvalidUploadGradingWorksheetPage(UploadException):
+    pass
+
+
+class InvalidUploadGradingConfirmationPage(UploadException):
+    pass
+
+
+def get_params_from_grading_page(
```

Comment:
`get_upload_grading_context`?

Reply:
Same thing here, I'm not sure about that. It is two HTML form parameter called `contextid` and `sesskey`
maybe `get_contextid_and_sesskey_params` ?